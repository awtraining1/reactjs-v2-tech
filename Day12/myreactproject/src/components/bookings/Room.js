import React from 'react';
import { useSearchParams } from 'react-router-dom';

const Room = (props) => {

    let [searchParams] = useSearchParams();

    let imgSrc = "https://cf.bstatic.com/xdata/images/hotel/270x200/269800030.jpg?k=8567acf226206bb28c9ff263d1c1934b462fecbb0d1bd360b85c2c59016585a4&o=";


    return (

        <div style={{ border: '5px solid green' }}>
            <center>
                <img src={imgSrc} width="200" height="150" />
            </center>

        Booking Room No
            { searchParams.get("roomNo") &&
                <b> {searchParams.get("roomNo")} </b>
            }

            {searchParams.get("days") &&
                <>for <b> {searchParams.get("days")} </b> days..</>
            }


        </div>

    )
}

export default Room;