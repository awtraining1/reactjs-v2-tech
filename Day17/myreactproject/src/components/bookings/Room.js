import React from 'react';
import { useSearchParams } from 'react-router-dom';
import { useState } from "react";

const Room = (props) => {

    let [state, setState] = useState({
        noOfdays:0
    });

    const handleChange = (event) => {
        setState(
            {
                ...state,
                [event.target.name]: event.target.value
            }
        );
    }

    let [searchParams] = useSearchParams();

    let imgSrc = "https://cf.bstatic.com/xdata/images/hotel/270x200/269800030.jpg?k=8567acf226206bb28c9ff263d1c1934b462fecbb0d1bd360b85c2c59016585a4&o=";


    return (

        <div style={{ border: '5px solid green' }}>
            <center>
                <img src={imgSrc} width="200" height="120" />
            </center>

        Booking Room NUMBER
            { searchParams.get("roomNo") &&
                <b> {searchParams.get("roomNo")} </b>
            }

            {searchParams.get("days") &&
                <>for <b> {searchParams.get("days")} </b> days..</>
            }

            <input type="text" name="noOfdays" 
            value={state.noOfdays}
            onChange={event => handleChange(event)} />

        </div>

    )
}

export default Room;