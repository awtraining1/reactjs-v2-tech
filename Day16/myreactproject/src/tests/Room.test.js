import { render, screen } from '@testing-library/react';
import user from "@testing-library/user-event";
import Room from '../components/bookings/Room'; 
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';  // Jest

// test('Room Test Case 1 textfeild input', () => {
//     render(<BrowserRouter><Room /></BrowserRouter>);
//     const t1 = screen.getByRole('textbox');  
    
//     user.type("5");
//     expect(screen.getByRole('textbox')).toHaveValue("5");
// });

test('Room Test Case 1 textfeild input', () => {
    render(<BrowserRouter><Room /></BrowserRouter>);
    
    const f1 = jest.fn();

    expect(screen.getByRole('textbox')).toHaveValue("5");
});

//Snapshot Test
test('2 Snapshot Test', () => {
    const tree = 
    renderer.create( <BrowserRouter><Room /></BrowserRouter>).toJSON();

    expect(tree).toMatchSnapshot();
});

//Enzyme is another React Testing library
// RSW - mock server for React Tests
