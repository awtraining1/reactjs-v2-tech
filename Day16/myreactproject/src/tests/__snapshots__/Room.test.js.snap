// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room Test Case 2 Snapshot Testing 1`] = `
<div
  style={
    Object {
      "border": "5px solid green",
    }
  }
>
  <center>
    <img
      height="120"
      src="https://cf.bstatic.com/xdata/images/hotel/270x200/269800030.jpg?k=8567acf226206bb28c9ff263d1c1934b462fecbb0d1bd360b85c2c59016585a4&o="
      width="200"
    />
  </center>
  Booking Room NUMBER
  <input
    name="noOfdays"
    onChange={[Function]}
    type="text"
    value={0}
  />
</div>
`;
